system/Database/Postgre/Connection.php

```diff
@@ -149,8 +149,8 @@ private function convertDSN()
      */
     public function reconnect()
     {
-        if (pg_ping($this->connID) === false) {
-            $this->connID = false;
+        if ($this->connID === false || pg_ping($this->connID) === false) {
+            $this->connect();
```

Comment:
Actually, I think it would be better to use the `initialize()` method because it will take into account all the settings (including failover). 
```suggestion
            $this->close();
            $this->initialize();
```

Reply:
Done.